tapioca/adapters.py

```diff
@@ -109,3 +111,36 @@ def format_data_to_request(self, data):
     def response_to_native(self, response):
         if response.content.strip():
             return response.json()
+
+
+class XMLAdapterMixin(object):
+
+    def _input_branches_to_xml_bytestring(self, data):
+        if isinstance(data, Mapping):
+            # removes the xml header and newline that xmltodict.unparse() adds
+            return xmltodict.unparse(data).split('\n')[1:][0].encode('utf-8')
+        try:
+            return data.encode('utf-8')
+        except Exception as e:
+            raise type(e)('Format not recognized, please enter an XML as string or a dictionary'
+                          'in xmltodict spec: \n%s' % e.message)
+
+    def get_request_kwargs(self, api_params, *args, **kwargs):
+        arguments = super(XMLAdapterMixin, self).get_request_kwargs(
+            api_params, *args, **kwargs)
+
+        if 'headers' not in arguments:
+            # allows user to override for formats like 'application/atom+xml'
+            arguments['headers'] = {}
+            arguments['headers']['Content-Type'] = 'application/xml'
```

Comment:
Good point, that is a better solution. I've made the change.

With line 135 inside the if statement, headers could be overridden at runtime like this (if no super calls set the headers) - this was the use case I was going after:

ipython console:
```
a = api.manage_worksheets(key='my_key').post(
    data=x, headers={'Content-Type': 'application/atom+xml'})
```

With the change, it's up to the wrapper developer to ensure that `get_request_kwargs` correctly defines the headers, and the user of the api wrapper won't have to think about it:

tapioca_gheets.py:
```
params['headers'] = {'Content-Type': 'application/atom+xml'}
```